lxd/task/schedule.go

```diff
@@ -0,0 +1,64 @@
+package task
+
+import (
+	"fmt"
+	"time"
+)
+
+// Schedule captures the signature of a schedule function.
+//
+// It should return the amount of time to wait before triggering the next
+// execution of a task function.
+//
+// If it returns zero, the function does not get run at all.
+//
+// If it returns a duration greater than zero, the task function gets run once
+// immediately and then again after the specified amount of time. At that point
+// the Task re-invokes the schedule function and repeats the same logic.
+//
+// If ErrSkip is returned, the immediate execution of the task function gets
+// skipped, and it will only be possibly executed after the returned interval.
+//
+// If any other error is returned, the task won't execute the function, however
+// if the returned interval is greater than zero it will re-try to run the
+// schedule function after that amount of time.
+type Schedule func() (time.Duration, error)
+
+// ErrSkip is a special error that maybe returned by a Schedule function to
+// mean to skip a particular execution of the task function, and just wait the
+// returned interval before re-evaluating.
+var ErrSkip = fmt.Errorf("skip execution of task function")
+
+// Every returns a Schdules that always returns the given time interval.
```

Comment:
Schedule

Reply:
Fixed in f8e6df366c5a18ba7dcbd0a3a72e05023162fc3d too.